### PR TITLE
chore: updates documentation for MatchMediaStylesheetBehavior changes

### PR DESCRIPTION
**Pull request by nicholasrice**
# Description
#3445 refactored `MatchMediaStylesheetBehavior` but did not update the docs site to reflect deprecation of `matchMediaStylesheetBehaviorFactory`.  This PR updates the documentation for the new feature and usage patterns.

## Issue type checklist

- [x] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [ ] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

**Adding or modifying component(s) in `@microsoft/fast-components` checklist**

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

## Process & policy checklist

- [ ] I have added tests for my changes.
- [ ] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/en/contributing/standards) for this project.

### CODE REVIEW

File: packages/web-components/fast-components/docs/design/match-media-stylesheets.md [hunk 1]
```diff
@@ -8,17 +8,41 @@ FAST exposes a mechanism to attach stylesheets conditionally based on a [MatchMe
 
 ### MatchMedia stylesheets
 
-`matchMediaStylesheetBehaviorFactory` can be used to construct a Behavior that will conditionally attach stylesheets based on the `matches` property of a [MediaQueryList](https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList).
+`matchMediaStylesheetBehavior` can be used to attach a stylesheet when a [MediaQueryList](https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList) matches and detach it when the query un-matches.
 
-__Example: Using the `matchMediaStylesheetBehaviorFactory`__
+__Example: Constructing the `MatchMediaStyleSheetBehavior`__
+```ts
```
Comment (janechu) [inline packages/web-components/fast-components/docs/design/match-media-stylesheets.md:14]: nit: this doc seems to be mix and matching the use of ts/js on the language selection for these codeblocks, should we just use either js or ts?

File: packages/web-components/fast-components/docs/design/match-media-stylesheets.md [hunk 2]
```diff
@@ -8,17 +8,41 @@ FAST exposes a mechanism to attach stylesheets conditionally based on a [MatchMe
 
 ### MatchMedia stylesheets
 
-`matchMediaStylesheetBehaviorFactory` can be used to construct a Behavior that will conditionally attach stylesheets based on the `matches` property of a [MediaQueryList](https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList).
+`matchMediaStylesheetBehavior` can be used to attach a stylesheet when a [MediaQueryList](https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList) matches and detach it when the query un-matches.
```
Comment (janechu) [inline packages/web-components/fast-components/docs/design/match-media-stylesheets.md:11]: should this be `matchMediaStylesheetBehavior ` or `MatchMediaStyleSheetBehavior`?

File: packages/web-components/fast-components/docs/design/match-media-stylesheets.md [hunk 3]
```diff
@@ -8,17 +8,41 @@ FAST exposes a mechanism to attach stylesheets conditionally based on a [MatchMe
...
+import { MatchMediaStyleSheetBehavior } from "@microsoft/fast-foundation";
+
+const mobileStylesheetBehavior = MatchMediaStyleSheetBehavior.with(
+    Window.matchMedia('(max-width: 600px)')
+);
 const styles = css`
     /* ... */
 `.withBehaviors(
-    maxWidthStylesheetBehavior(css`
+    mobileBehavior(css`
```
Comment (chrisdholt) [inline packages/web-components/fast-components/docs/design/match-media-stylesheets.md:45]: should this match the const on line 39?

File: packages/web-components/fast-components/docs/design/match-media-stylesheets.md (hunk 2 above)
Comment (nicholasrice) [reply packages/web-components/fast-components/docs/design/match-media-stylesheets.md]: `M` - thanks

File: packages/web-components/fast-components/docs/design/match-media-stylesheets.md (hunk 3 above)
Comment (nicholasrice) [reply packages/web-components/fast-components/docs/design/match-media-stylesheets.md]: sure should - good catch

**Pull request merged**